test/e2e/common.go

```diff
@@ -125,19 +125,24 @@ func getSha256Hash(filename string) ([]byte, error) {
 	return hash.Sum(nil), nil
 }
 
-func DumpSpecResourcesAndCleanup(ctx context.Context, specName string, clusterProxy framework.ClusterProxy, artifactFolder string, namespace string, intervalsGetter func(spec, key string) []interface{}, clusterName, clusterctlLogFolder string, skipCleanup bool) {
+func DumpSpecResourcesAndCleanup(ctx context.Context, specName string, bootstrapClusterProxy framework.ClusterProxy, targetClusterProxy framework.ClusterProxy, artifactFolder string, namespace string, intervalsGetter func(spec, key string) []interface{}, clusterName, clusterctlLogFolder string, skipCleanup bool) {
 	Expect(os.RemoveAll(clusterctlLogFolder)).Should(Succeed())
-	clusterClient := clusterProxy.GetClient()
+	clusterClient := bootstrapClusterProxy.GetClient()
 
-	clusterProxy.CollectWorkloadClusterLogs(ctx, namespace, clusterName, artifactFolder)
+	bootstrapClusterProxy.CollectWorkloadClusterLogs(ctx, namespace, clusterName, artifactFolder)
 
+	By("Fetch logs from target cluster")
+	err := FetchClusterLogs(targetClusterProxy, clusterLogCollectionBasePath)
+	if err != nil {
+		Logf("Error: %v", err)
+	}
 	// Dumps all the resources in the spec namespace, then cleanups the cluster object and the spec namespace itself.
 	By(fmt.Sprintf("Dumping all the Cluster API resources in the %q namespace", namespace))
 	// Dump all Cluster API related resources to artifacts before deleting them.
 	framework.DumpAllResources(ctx, framework.DumpAllResourcesInput{
 		Lister:    clusterClient,
 		Namespace: namespace,
-		LogPath:   filepath.Join(artifactFolder, "clusters", clusterProxy.GetName(), "resources"),
+		LogPath:   filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName(), "resources"),
```

Comment:
If we have `target_clusters` log dir, shouldn't this be `bootstrap_cluster` (instead of `clusters`, we also only have 1 bootstrap cluster, I suppose?)

And do we need the cluster name?

Reply:
Here `taget_cluster_logs` log directory was used to align the previous format of log collection and from project infra it is looking for same directory and copy it from here.  

Also I think `bootstrapClusterProxy ` is just a rename of `clusterProxy`, In remediation it involves the bootstrap cluster.

Reply:
Just "resources" is fine here we don't need to mention the cluster name. I have changed it. please check.